src/Arcus.BackgroundJobs.DataBricks/DatabricksInfoProvider.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using GuardNet;
+using Microsoft.Azure.Databricks.Client;
+
+namespace Arcus.BackgroundJobs.DataBricks
+{
+    /// <summary>
+    /// Provides dev-friendly access to the Databricks instance.
+    /// </summary>
+    internal class DatabricksInfoProvider : IDisposable
+    {
+        private readonly DatabricksClient _databricksClient;
+
+        /// <summary>
+        /// Initializes a new instance of hte <see cref="DatabricksInfoProvider"/> class.
+        /// </summary>
+        /// <param name="databricksClient">The client to interact with Databricks.</param>
+        internal DatabricksInfoProvider(DatabricksClient databricksClient)
+        {
+            Guard.NotNull(databricksClient, nameof(databricksClient));
+
+            _databricksClient = databricksClient;
+        }
+
+        /// <summary>
+        /// Get all job runs that were finished in a given time window
+        /// </summary>
+        /// <param name="startOfWindow">Start of time window which we are interested in. (Inclusive)</param>
+        /// <param name="endOfWindow">End of time window which we are interested in. (Exclusive)</param>
+        internal async Task<IEnumerable<(string jobName, Run jobRun)>> GetFinishedJobRunsAsync(DateTimeOffset startOfWindow, DateTimeOffset endOfWindow)
```

Comment:
Let's create an object for this?

Reply:
Yes, not sure. I like tuples, especially in internal things. But that can be a F#-ish thing of mine.

Reply:
I'd like to open it up, see above